doc/devel/codespaces.md

```diff
@@ -0,0 +1,107 @@
+# Contributing to Matplotlib using GitHub codespaces
+
+You've discovered a bug or something else you want to change
+in Matplotlib — excellent!
+
+You've worked out a way to fix it — even better!
+
+You want to tell us about it — best of all!
+
+This project is a community effort, and everyone is welcome to contribute.
+Everyone within the community is expected to abide by our
+[Code of Conduct](../../CODE_OF_CONDUCT.md).
+
+## GitHub codespaces contribution workflow
+
+The preferred way to contribute to Matplotlib is to fork the main
+repository at https://github.com/matplotlib/matplotlib, then submit a "pull
+request" (PR). You can do this by cloning a copy of the Maplotlib repository to
+your own computer, or alternatively using
+[GitHub Codespaces](https://docs.github.com/codespaces) (a cloud-based
+in-browser development environment, that comes with the appropriated setup to
+contribute to Matplotlib).
+
+A brief overview of the workflows is as follows.
+
+1. Go to the GitHub web interface
+2. Fork the [project repository](https://github.com/matplotlib/matplotlib):
+3. Open codespaces on your fork by clicking on the green "Code" button on the
+   GitHub web interface and selecting the "Codespaces" tab. Next, click on "Open
+   codespaces on <your fork name>". You will be able to change branches later,
+   so you can select the default `main` branch.
+
+   After the codespace is created, you will be taken to a new browser tab where
+   you can use the terminal to activate a pre-defined conda environment called
+   `mpl-dev`:
+
+   ```
+   conda activate mpl-dev
```

Comment:
I think there is a vscode setting file we can drop to tell vscode we want to use the environment and it will auto-activate for us.

Reply:
I will investigate but this was not possible until recently because of the way the terminal is fired up when codespaces is first opened. They have been making a lot of updates lately so I'll verify.